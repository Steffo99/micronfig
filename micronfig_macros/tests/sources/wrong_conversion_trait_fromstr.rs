micronfig::config! {
	GARASAUTO: String > std::convert::Infallible,
}

fn main() {
	std::env::set_var("GARASAUTO", "!");
	println!("{:#?}", GARASAUTO());
}
