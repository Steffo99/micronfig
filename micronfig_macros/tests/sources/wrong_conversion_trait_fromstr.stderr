error[E0277]: the trait bound `Infallible: FromStr` is not satisfied
 --> tests/sources/wrong_conversion_trait_fromstr.rs:1:1
  |
1 | / micronfig::config! {
2 | |     GARASAUTO: String > std::convert::Infallible,
3 | | }
  | |_^ the trait `FromStr` is not implemented for `Infallible`
  |
  = help: the following other types implement trait `FromStr`:
            IpAddr
            Ipv4Addr
            Ipv6Addr
            NonZero<i128>
            NonZero<i16>
            NonZero<i32>
            NonZero<i64>
            NonZero<i8>
          and $N others
note: required by a bound in `core::str::<impl str>::parse`
 --> $RUST/core/src/str/mod.rs
  |
  |     pub fn parse<F: FromStr>(&self) -> Result<F, F::Err> {
  |                     ^^^^^^^ required by this bound in `core::str::<impl str>::parse`
  = note: this error originates in the macro `micronfig::config` (in Nightly builds, run with -Z macro-backtrace for more info)
